CDCgov/prime-reportstream
# Flexion/jgilmore/update blob storage transport filenames

This PR updates Blob Storage Transport to utilize the file-naming-templates in the same manner as the current SFTP transport. Currently Blob Storage Transport names the file copied into the receiver's storage container with the same name as the file in the RS "ready" folder. This will impact existing receivers utilizing blob storage transport so it may be worth adding a file-naming-template that simply passes along the existing filename. ETOR

Test Steps:
1. run ./prime test

2. run ./gradlew testIntegration

3. Add the attached settings to local organizations.yml file
[golden-copy-test-settings.txt](https://github.com/user-attachments/files/18264051/golden-copy-test-settings.txt)

4. In storage account devstoreaccount1 create a container named "test"

5. Use attached test message to send a message to local reportstream from sender "flexion.simulated-hospital"
[test-blob-name.txt](https://github.com/user-attachments/files/18264103/test-blob-name.txt)


6. Check created "test" container for two files, one named "golden-copy-<UUID>.hl7" and one named "oru_r01-base.yml-<UUID>.hl7

## Changes
- simplify copyBlob function to only return the byteArray of the copied blob
- add logic to utilize the externalFilename in blobstoretransport
- clean tests for copyBlob


### Testing
- [x ] Tested locally?
- [x ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [x ] Added tests?
